minetest/minetest
# Add a setting to group transparency sorted triangles by buffer

Workaround for #12900.

Adds setting `transparency_sorting_group_by_buffers`.
(Default on, because I think this is more important for the average user than correct transparency sorting.)

If on, triangles in the same buffer are batched into the same PartialMeshBuffer, resulting in fewer draw calls, but wrong order.
Triangles within a single buffer are still ordered, and buffers are ordered relative to each other (with respect to their nearest triangle).

## To do

This PR is a Ready for Review.

## How to test

* Check if #12900 is fast now.
* Use different blended transparent nodes to test transparency sorting.
  Example setup (the marble in the back is invisible because the marble in the front has to be drawn in front of the water):
  ![screenshot_20240903_183352](https://github.com/user-attachments/assets/6dced24c-0543-48f9-b37c-15fcd0a442bc)
